pkg/doc/did/doc.go

```diff
@@ -112,7 +112,12 @@ type PublicKey struct {
 	ID         string
 	Type       string
 	Controller string
-	Value      []byte
+
+	Value []byte
+
+	Algorithm string `json:"-"`
```

Comment:
Why we need to show Alg, Curve & KeyType in exported did.Doc.PublicKey? Why can't we just decode public key internally and stick to old structure?

Reply:
I think we can at least make it hidden, why not?

Reply:
Those new fields are made private, thank you @sudeshrshetty

Reply:
looks good, but what is the use of those private fields?
 As I can see we are setting it from jwk instance but not being used anywhere afterwards.

Reply:
usually `Curve` is exposed (like in ecdsa.PublicKey)

Reply:
Why clients should care about curve? For key type based logics - they should be able to figure out using public key type

Reply:
For `JsonWebSignature2020`, it's not enough to have only `type` value of the public key. Look at the example:
```
  "publicKey": [
    {
      "id": "did:example:123#_Qq0UL2Fq651Q0Fjd6TvnYE-faHiOpRlPVQcY_-tA4A",
      "type": "JwsVerificationKey2020",
      "controller": "did:example:123",
      "publicKeyJwk": {
        "crv": "Ed25519",
        "x": "VCpo2LMLhn6iWku8MKvSLg2ZAoC-nlOyPVQaO3FxVeQ",
        "kty": "OKP",
        "kid": "_Qq0UL2Fq651Q0Fjd6TvnYE-faHiOpRlPVQcY_-tA4A"
      }
    },
    {
      "id": "did:example:123#4SZ-StXrp5Yd4_4rxHVTCYTHyt4zyPfN1fIuYsm6k3A",
      "type": "JwsVerificationKey2020",
      "controller": "did:example:123",
      "publicKeyJwk": {
        "crv": "secp256k1",
        "x": "Z4Y3NNOxv0J6tCgqOBFnHnaZhJF6LdulT7z8A-2D5_8",
        "y": "i5a2NtJoUKXkLm6q8nOEu9WOkso1Ag6FTUT6k_LMnGk",
        "kty": "EC",
        "kid": "4SZ-StXrp5Yd4_4rxHVTCYTHyt4zyPfN1fIuYsm6k3A"
      }
    }
  ]
```

The `type` is the same and is `JwsVerificationKey2020`. 
We can't deduce is it `Ed25519` or `secp256k1` etc. We need to extract extra information from `publicKeyJwk`.
That's the problem solved in this PR.